mini-rx2/src/main/java/mini/rx/RxUtils.kt

```diff
@@ -57,20 +57,18 @@ class DefaultSubscriptionTracker : SubscriptionTracker {
     }
 }
 
-fun <S> Store<S>.flowable(hotStart: Boolean = true): Flowable<S> {
-    val processor = PublishProcessor.create<S>()
-    val subscription = subscribe(hotStart = false) {
-        processor.offer(it)
-    }
-    return processor.doOnTerminate { subscription.close() }
-        .let { if (hotStart) it.startWith(state) else it }
-}
-
 fun <S> Store<S>.observable(hotStart: Boolean = true): Observable<S> {
     val subject = PublishSubject.create<S>()
     val subscription = subscribe(hotStart = false) {
         subject.onNext(it)
     }
-    return subject.doOnTerminate { subscription.close() }
+    return subject
+        .doOnDispose { subscription.close() }
+        .doOnTerminate { subscription.close() }
         .let { if (hotStart) it.startWith(state) else it }
+}
```

Comment:
I'll add test for `observable` extension too

Reply:
It's already there 😛